django_irods/views.py

```diff
@@ -0,0 +1,477 @@
+    # ))
+
+    # at this point, we've modified the output_path and irods_output_path for three cases.
+    # aggregation logic only applies if the download request isn't a bag, zipfile, or folder,
+    # and the thing to be downloaded is a "single file aggregation" object.
+    if not is_bag_download and not is_zip_download and not is_zip_request and \
+       res.resource_type == 'CompositeResource':
+        if 'data/contents/' in path:  # not a metadata file
+            for f in ResourceFile.objects.filter(object_id=res.id):
+                if path == f.storage_path:
+                    if f.has_logical_file and f.logical_file.is_single_file_aggregation:
+                        is_sf_agg_file = True
+                        daily_date = datetime.datetime.today().strftime('%Y-%m-%d')
+                        output_path = "zips/{}/{}/{}.zip".format(daily_date, uuid4().hex, path)
+                        if res.is_federated:
+                            irods_output_path = os.path.join(res.resource_federation_path,
+                                                             output_path)
+                        else:
+                            irods_output_path = output_path
+                break
+
+    # After this point, we have valid path, irods_path, output_path, and irods_output_path
+    # At most one of the following flags has also been set:
+    # * is_bag_download: download a bag in format bags/{rid}.zip
+    # * is_zip_download: download a zipfile in format zips/{date}/{random guid}/{path}.zip
+    # * is_zip_request: path is a folder; zip before returning
+    # * is_sf_agg_file: path is a single-file aggregation in Composite Resource, return a zip
+    # if none of these are set, it's a normal download
+
+    # logger.debug(" after aggregation check, bag={}, zipd={} zipr={} sfagg={}".format(
+    #     str(is_bag_download),
+    #     str(is_zip_download),
+    #     str(is_zip_request),
+    #     str(is_sf_agg_file)
+    # ))
+    # logger.debug(" after aggregation check, path={}, opath={}".format(
+    #     path, output_path
+    # ))
+
+    # determine active session
+    if res.is_federated:
+        # the resource is stored in federated zone
+        session = icommands.ACTIVE_SESSION
+    else:
+        # TODO: From Alva: I do not understand the use case for changing the environment.
+        # TODO: This seems an enormous potential vulnerability, as arguments are
+        # TODO: passed from the URI directly to IRODS without verification.
+        if 'environment' in kwargs:
+            logger.warn("setting iRODS from environment")
+            environment = int(kwargs['environment'])
+            environment = m.RodsEnvironment.objects.get(pk=environment)
+            session = Session("/tmp/django_irods", settings.IRODS_ICOMMANDS_PATH,
+                              session_id=uuid4())
+            session.create_environment(environment)
+            session.run('iinit', None, environment.auth)
```

Comment:
I agree and I don't think this conditional branch has ever been executed since we never pass an "environment" in kwargs. I'd suggest to remove this conditional branch.